src/engine/app.rs

```diff
@@ -16,11 +16,10 @@ pub struct Application {
 }
 
 impl Application {
-    /// Creates a new Application with the given initial game state, planner, and config.
-    pub fn new<T>(initial_state: T, planner: Planner<Arc<Mutex<Context>>>, config: Config) -> Application
+    /// Creates a new Application with the given initial game state, planner, and context.
+    pub fn new<T>(initial_state: T, planner: Planner<Arc<Mutex<Context>>>, context: Context) -> Application
```

Comment:
`Planner` doesn't own an `Arc<Mutex<Context>>` it just denotes the type of `Planner::dispatch` argument and the type of `Processor::run` argument.

Passing `Planner` to `Application::new` allows adding processors to planner and registering components before it is consumed by `Application::new`, `ApplicationBuilder::done` calls it.